File: Video Chat Pro ActiveX Control/Examples/vb.net2015/Server/Form1.vb
Public Class Form1

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If Me.textPort.Text = "" Then
            MessageBox.Show("Plesase input Port No")
            Exit Sub
        End If
        Me.AxVideoChatServer1.InitServer(Convert.ToInt16(textPort.Text), 300)
        If Me.CheckBox1.Checked = True Then
            Me.AxVideoChatServer1.UseRandomNumber = True
        Else
            Me.AxVideoChatServer1.UseRandomNumber = False
        End If
        Me.textConference.Text = Me.AxVideoChatServer1.CreateConference
        Me.listUserID.Items.Clear()
        Me.Button1.Enabled = False
        Me.Button2.Enabled = True
    End Sub

    Private Sub Form1_FormClosed(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        Me.AxVideoChatServer1.UnInitServer()

    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.Button1.Enabled = True
        Me.Button2.Enabled = False
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        Dim userID As Integer
        userID = Me.AxVideoChatServer1.AddUser(Me.textConference.Text)
        If userID <> 0 Then
            Me.listUserID.Items.Add(userID)
        Else
            MessageBox.Show("Max is 12 user per Room")
        End If

    End Sub

    Private Sub listUserID_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles listUserID.Click
        Me.textSelectedUserID.Text = Me.listUserID.Text
    End Sub

    Private Sub listUserID_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles listUserID.SelectedIndexChanged

    End Sub



    Private Sub AxVideoChatServer1_ClientConnected(ByVal sender As Object, ByVal e As AxVideoChatServerLib._DVideoChatServerEvents_ClientConnectedEvent) Handles AxVideoChatServer1.ClientConnected
        Dim iCount As Short
        Dim strUserID As String
        Dim strListID As String


        iCount = Me.listUserID.Items.Count - 1
        strUserID = e.iUserID.ToString
        For i = 0 To iCount
            strListID = Me.listUserID.Items(i)

            If Trim(strListID) = Trim(strUserID) Then
                Me.listUserID.Items.RemoveAt(i)
                Me.listUserID.Items.Insert(i, strUserID + "***")
                'Me.listUserID.Items(i) = Me.listUserID.Items.Item(i).ToString + "***"
            End If

        Next

        Me.listUserID.Refresh()
    End Sub

    Private Sub AxVideoChatServer1_ClientDisconnected(ByVal sender As Object, ByVal e As AxVideoChatServerLib._DVideoChatServerEvents_ClientDisconnectedEvent) Handles AxVideoChatServer1.ClientDisconnected
        Dim iCount As Short
        Dim strUserID As String
        Dim strListID As String
        iCount = Me.listUserID.Items.Count - 1
        strUserID = e.iUserID.ToString
        For i = 0 To iCount
            strListID = Me.listUserID.Items(i)

            If strListID = (strUserID + "***") Then
                Me.listUserID.Items.RemoveAt(i)
                Me.listUserID.Items.Insert(i, strUserID)
            End If

        Next
        Me.listUserID.Refresh()
    End Sub
End Class

File: Video Chat Pro ActiveX Control/Examples/c#2015/Server/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            if (this.textPort.Text == "")
            {
                MessageBox.Show("Please input Port No.");
                return;
            }
            this.axVideoChatServer1.InitServer(Convert.ToInt32(this.textPort.Text), 300);
            
            if (this.chkUseRandomNo.Checked == true)
            {
                this.axVideoChatServer1.UseRandomNumber = true;
            }
            else
            {
                this.axVideoChatServer1.UseRandomNumber = false;
            }

            
            this.textConfNo.Text = Convert.ToString(this.axVideoChatServer1.CreateConference());
            this.listBox1.Items.Clear();
         
        
            this.button1.Enabled = false;
            this.button2.Enabled = true;
        }

        
        private void Form1_Load(object sender, EventArgs e)
        {
            this.button1.Enabled = true;
            this.button2.Enabled = false;
        }

        private void Form1_Closing(object sender, System.ComponentModel.CancelEventArgs e)
        {

            this.axVideoChatServer1.UnInitServer();


        }


        private void button2_Click(object sender, EventArgs e)
        {
            long iUserID = 0;
            iUserID = this.axVideoChatServer1.AddUser(Convert.ToInt32(this.textConfNo.Text));

            

            if (iUserID != 0)
            {
                this.listBox1.Items.Add(iUserID.ToString());
            }
            else
            {
                MessageBox.Show("Max is 12 user per Room");
            }
        }

        private void listUserID_SelectedIndexChanged(object sender, EventArgs e)
        {
        }

        private void axVideoChatServer1_ClientConnected(object sender, AxVideoChatServerLib._DVideoChatServerEvents_ClientConnectedEvent e)
        {
            string strUserID;
            string strListID;
            
            strUserID = e.iUserID.ToString();
            
            
            int index = listBox1.Items.IndexOf(strUserID);
            if (index != -1)
            {
                listBox1.Items.RemoveAt(index);
                listBox1.Items.Insert(index, strUserID+"***");
                listBox1.SelectedIndex = index;
            }

            this.listBox1.Refresh();
        }

        private void axVideoChatServer1_ClientDisConnected(object sender, AxVideoChatServerLib._DVideoChatServerEvents_ClientDisconnectedEvent e)
        {
            string strUserID;
            string strListID;
            strUserID = e.iUserID.ToString();



            for (int i = 0; i < this.listBox1.Items.Count; i++)
            {
                strListID = this.listBox1.Items[i].ToString();
                if (strListID == (strUserID + "***"))
                {
                    this.listBox1.Items.RemoveAt(i);
                    this.listBox1.Items.Insert(i, strUserID);
                   }
            }

            this.listBox1.Refresh();
        }
       

        private void button3_Click(object sender, EventArgs e)
        {

            
        }

        private void button4_Click(object sender, EventArgs e)
        {
        }

        private void listBox1_Click(object sender, EventArgs e)
        {
            textSelectedUserID.Text = this.listBox1.Text;
        }

        private void listBox1_SelectedIndexChanged(object sender, EventArgs e)
        {

        }

        private void chkUseRandomNo_CheckedChanged(object sender, EventArgs e)
        {

        }

        private void button3_Click_1(object sender, EventArgs e)
        {
            int index = listBox1.Items.IndexOf(textSelectedUserID.Text);
            if (index != -1)
            {
                listBox1.Items.RemoveAt(index);
                listBox1.Items.Insert(index, textSelectedUserID.Text + "***");
                //listBox1.Items[index] = strUserID + "***";
                listBox1.SelectedIndex = index;
            }

        }

        private void timer1_Tick(object sender, EventArgs e)
        {

        }
    }
}
